src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -371,13 +458,89 @@ asf::auto_release_ptr<asr::Project> AppleseedInteractiveRender::prepare_project(
             renderer_settings,
             m_bitmap,
             time,
-            m_progress_cb));
+            m_progress_cb,
+            m_object_instance_map,
+            m_material_map));
 
     std::setlocale(LC_ALL, previous_locale.c_str());
 
     return project;
 }
 
+void AppleseedInteractiveRender::assign_material(Mtl* mtl, INode* node)
+{
+    if (mtl == nullptr)
+        return;
+
+    std::vector<Mtl*> materials;
+    const int submtlcount = mtl->NumSubMtls();
+    if (mtl->IsMultiMtl() && submtlcount > 0)
+    {
+        for (int i = 0; i < submtlcount; ++i)
+        {
+            Mtl* submtl = mtl->GetSubMtl(i);
+            materials.push_back(submtl);
+        }
+    }
+    else
+    {
+        materials.push_back(mtl);
+    }
+
+    MaterialMap assigned_materials;
+    for (const auto& mtl : materials)
+    {
+        IAppleseedMtl* appleseed_mtl =
```

Comment:
It looks like you don't need that variable, and you don't need the cast either. Maybe this would be sufficient:

    if (mtl->GetInterface(IAppleseedMtl::interface_id()) == nullptr)
        continue;